src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java

```diff
@@ -78,6 +79,44 @@ public void createPrimaryUserAffiliation(String eventPayload) {
     }
   }
 
+  @Override
+  @SneakyThrows
+  @Transactional
+  public void updatePrimaryUserAffiliation(String eventPayload) {
+    FolioExecutionContext currentContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    String centralTenantId = folioExecutionContext.getTenantId();
+    try {
+      var userEvent = objectMapper.readValue(eventPayload, UserEvent.class);
+      log.info("Received event for update primary affiliation for user: {} and tenant: {}", userEvent.getUserDto().getId(), userEvent.getTenantId());
+
+      var consortiaTenant = tenantService.getByTenantId(userEvent.getTenantId());
+      if (consortiaTenant == null) {
+        log.warn("Tenant {} not exists in consortia", userEvent.getTenantId());
+        return;
+      }
+      UUID userId = getUserId(userEvent);
+      String newUsername = userEvent.getUserDto().getUsername();
+
+      UserTenantCollection userTenantCollection = userTenantService.getByUserId(consortiaTenant.getConsortiumId(), userId, 0, Integer.MAX_VALUE);
+      boolean isUsernameChanged = userTenantCollection.getUserTenants().stream()
+        .anyMatch(userTenant -> ObjectUtils.notEqual(userTenant.getUsername(), newUsername));
+
+      if (isUsernameChanged) {
+        userTenantService.updateUsernameInPrimaryUserTenantAffiliation(userId, newUsername);
+        log.info("Username in primary affiliation has been updated for the user: {}", userEvent.getUserDto().getId());
+      }
+
+      PrimaryAffiliationEvent affiliationEvent = createPrimaryAffiliationEvent(userEvent);
+      String data = objectMapper.writeValueAsString(affiliationEvent);
+
+      try (var context = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentContext))) {
```

Comment:
isn't it central tenant folioExecutionContext already?

Reply:
Refactored